### PR TITLE
CircleCI: support for intergration branches

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -345,17 +352,26 @@ on-version-tags: &on-version-tags
       only: /^v[0-9].*/
 
 on-master-and-version-tags: &on-master-and-version-tags
+  filters:
+    branches:
+      only: master
+    tags:
+      only: /^v[0-9].*/
```
Comment (DvirDukhan) [inline .circleci/config.yml:359]: remove `on-master-and-version-tags` as it is no longer in use, or combine both definitions

File: .circleci/config.yml [hunk 2]
```diff
@@ -405,7 +421,7 @@ workflows:
           # <<: *on-version-tags
       - deploy-snapshot:
           <<: *after-platform-builds
-          <<: *on-master
+          <<: *on-integ-branch
```
Comment (DvirDukhan) [inline .circleci/config.yml:424]: why do we want `integ` branch snapshots? will it overwrite existing versions?

Comment (rafie) [reply .circleci/config.yml:424]: like we get a master snapshot, we may want integration branch snapshots, with corresponding names, like `redisai/snapshots/redisai-cpu-onnxruntime.linux-bionic-x64.1.2.tgz`.
we can limit ourselves to `on-master` at the moment.

File: .circleci/config.yml (hunk 1 above)
Comment (rafie) [reply .circleci/config.yml:359]: yes.

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**